### PR TITLE
Adding Turns

### PR DESCRIPTION
**Pull request by CodeZero13**
Turns have been implemented, allowing each team a single move before advancing to the next team's turn.  Will likely need to be changed later during conversion from chess.

### CODE REVIEW

File: src/Map/Turn.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ChessDemo.Entities;
+
+namespace ChessDemo.Map
+{
+    class Turn
```
Comment (ryzngard) [inline src/Map/Turn.cs]: Let's just call this TurnManager and make it a singleton

File: src/Map/Turn.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        /// List of Teams to cycle through
+        private List<Team> teamList;
+
+        /// Dictionary of teams to entities
+        private Dictionary<Entity, Team> teamDict;
+
+        /// Team which is currently active
+        public Team currentTeam;
+
+        public Turn(Dictionary<Entity, Team> teams)
```
Comment (ryzngard) [inline src/Map/Turn.cs]: With TurnManager being a singletong, make an Add(Dictionary<Entity, Team> team). Probably will also want Add(Entity, Team) so we can add something to a Team later. I think TurnManager can be the decisive factor for who is on what team.

File: src/Map/Turn.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            this.teamDict = teams;
+        }
+
+        public void initTeams() {
+
+            teamList = teamDict.Values.ToList();
+            currentTeam = teamList.First();
+        }
+
+        public Boolean myTurn(Entity currentEntity)
```
Comment (ryzngard) [inline src/Map/Turn.cs]: Nit: Generally in C# methods start with a capital letter. The only thing with lowercase start is private variables

Comment (ryzngard) [inline src/Map/Turn.cs]: Also, call this IsMyTurn(Entity currentEntity), since it makes it more of a question and less declarative.

**Comment by CodeZero13** [comment]
Added turn changes again after updating pulling in the dll changes.  Made TurnManager a singleton and it is now responsible for maintaining the list of Teams.  Still using the same logic as before, one action per turn, temporarily until we implement an in game menu or button to advance the turn and logic for each unit to exhaust their actions per turn.

**Pull request merged**